docs/src/literate/ns_vs_diffeq.jl

```diff
@@ -0,0 +1,507 @@
+# and $f$ the right hand side. For us $f$ can be seen as the spatial discretization
+# of all linear and non-linear operators depending on $u$ and $t$, but not on the time
+# derivative of $u$.
+#
+# ## Some Theory on the Incompressible Navier-Stokes Equations
+#
+# ### Problem Description in Strong Form
+#
+# The incompressible Navier-Stokes equations can be stated as the system
+#
+# ```math
+#  \begin{aligned}
+#    \partial_t v &= \underbrace{\nu \Delta v}_{\text{viscosity}} - \underbrace{(v \cdot \nabla) v}_{\text{advection}} - \underbrace{\nabla p}_{\text{pressure}} \\
+#               0 &= \underbrace{\nabla \cdot v}_{\text{incompressibility}}
+#  \end{aligned}
+# ```
+#
+# where $v$ is the unknown velocity field, $p$ the unknown pressure field,
+# $\nu$ the dynamic viscosity and $\Delta$ the Laplacian. In the derivation we assumed
+# a constant density of 1 for the fluid and negligible coupling between the velocity components.
+# Finally we see that the pressure term appears only in combination with the gradient
+# operator, so for any solution $p$ the function $p + c$ is also an admissible solution, if
+# we do not impose Dirichlet conditions on the pressure. To resolve this we introduce the
+# implicit constraint that $ \int_\Omega p = 0 $.
+#
+#
+# Our setup is derived from [Turek's DFG benchmark](http://www.mathematik.tu-dortmund.de/~featflow/en/benchmarks/cfdbenchmarking/flow/dfg_benchmark1_re20.html).
+# We model a channel with size $0.41 \times 2.2$ and a hole of radius $0.05$ centered at $(0.2, 0.2)$.
+# The left side has a parabolic inflow profile, which is ramped up over time, modeled as the time dependent
+# Dirichlet condition
+# ```math
+#  v(x,y,t)
+#  =
+#  \begin{bmatrix}
+#      4 v_{in}(t) \cdot y \cdot (0.41-y)/0.41^2 \\
+#      0
+#  \end{bmatrix}
+# ```
+# where $v_{in}(t) = \text{clamp}(t, 0.0, 1.0)$. With a viscosity of $\nu = 0.001$
+# this is enough to induce turbulence behind the cylinder which leads to vortex shedding. The top and bottom of our
+# channel have no-slip conditions, i.e. $v = [0,0]^T$, while the right boundary has the do-nothing boundary condtion
+# $\nu \partial_n v - p n = 0$ to model outflow. With these boundary conditions we can choose the zero solution as a
+# feasible initial condition.
+#
+# ### Derivation of Semi-Discrete Weak Form
+#
+# By multiplying test functions $\varphi$ and $\psi$ from a suitable test function space on the strong form,
+# followed by integrating over the domain and applying partial integration to the pressure and viscosity terms
+# we can obtain the following weak form
+# ```math
+#  \begin{aligned}
+#    \int \partial_t v \cdot \varphi &= - \int \nu \nabla v : \nabla \varphi - \int (v \cdot \nabla) v \cdot \varphi + \int p (\nabla \cdot \varphi) + \int_{\partial \Omega_{N}} \underbrace{(\nu \partial_n v - p n )}_{=0} \cdot \varphi \\
+#                                  0 &= (\int \nabla \cdot v) \psi
```

Comment:
```suggestion
#                                  0 &= \left(\int \nabla \cdot v \right) \psi
```
not necessary but it's r/mildlyinfuriating without `\left\right`:D

Reply:
oh wait, is the opening bracket actually intended before the integral? Or should it be in front of the nabla?

Reply:
The integral should be on the outside